src/context/CaptureDetailContext.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, createContext } from 'react';
+import api from '../api/treeTrackerApi';
+import * as loglevel from 'loglevel';
+
+const log = loglevel.getLogger('../context/CaptureDetailContext');
+
+export const CaptureDetailContext = createContext({
+  capture: null,
+  species: null,
+  tags: [],
+  getCaptureDetail: () => {},
+  getCapture: () => {},
+  getSpecies: () => {},
+  getTags: () => {},
+  reset: () => {},
+});
+
+const STATE_EMPTY = {
+  capture: null,
+  species: null,
+  tags: [],
+};
+
+export function CaptureDetailProvider(props) {
+  const [state, setState] = useState(STATE_EMPTY);
+
+  // STATE HELPER FUNCTIONS
+
+  function reset() {
+    setState(STATE_EMPTY);
+  }
+
+  // EVENT HANDLERS
+
+  const getCaptureDetail = async (id) => {
+    reset();
+
+    return getCapture(id).then((capture) => {
+      getSpecies(capture && capture.speciesId);
+      getTags(capture && capture.treeTags);
+    });
+  };
+
+  const getCapture = async (id) => {
+    if (id == null) {
+      log.debug('getCapture called with no id');
+      return Promise.resolve(STATE_EMPTY.capture);
+    }
+
+    return api.getCaptureById(id).then((capture) => {
+      setState({ ...state, capture });
+    });
+  };
+
+  const getSpecies = async (speciesId) => {
+    if (speciesId == null) {
+      return Promise.resolve(STATE_EMPTY.species);
+    }
+
+    return api.getSpeciesById(speciesId).then((species) => {
+      setState({ ...state, species });
+    });
+  };
```

Comment:
@gwynndp I am not sure I'm correct, maybe context approach has some downsides like: if using Redux, we can reuse this piece of code by put it into the species module/action/reducer, so we can invoke species.dispatch.getSpeciesByID here?

Reply:
Context and redux do have different strengths and weaknesses. Context is good for holding state that might be used by a particular tree of components.  The .getSpeciesById() can be called without context if the api file is imported, so I'm not sure there is a limitation in this case.  But, I do think that we might review the organization of data into the contexts.  I kept it close to the way it was organized in the stores wit the refactor, but it might not hurt to allow verify context to hold data for tags and species if it helps simplify things, for example.   

There are some changes already planned for the admin panel that may make these concerns mute, so we should revisit this discussion later.